tools/scripts/replace_static.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+# Usage:
+#   ./replace_static_functions.sh <source_file> [--replace-vars]
+#
+# - Replaces all 'static' function definitions with non-static.
+# - Replaces all 'static inline' with 'inline'.
+# - If --replace-vars is given, also replaces 'static' at variable definitions.
+
+set -e
+
+SOURCE_FILE="$1"
+REPLACE_VARS=0
+
+if [[ "$2" == "--replace-vars" ]]; then
+  REPLACE_VARS=1
+fi
+
+if [[ ! -f "$SOURCE_FILE" ]]; then
+  echo "Source file '$SOURCE_FILE' not found!"
+  exit 1
+fi
+
+TMP_FILE="${SOURCE_FILE}.tmp.$$"
+
+echo "[INFO] Replacing 'static inline' with 'inline' in $SOURCE_FILE"
+sed -E 's/\bstatic[[:space:]]+inline\b/inline/g' "$SOURCE_FILE" > "$TMP_FILE"
+
+echo "[INFO] Replacing 'static' at function start in $SOURCE_FILE"
+# This matches 'static' at the start of a line (possibly with spaces), followed by anything, up to '('
+sed -E -i 's/^([[:space:]]*)static([[:space:]]+)([a-zA-Z_][a-zA-Z0-9_:<>~ \*&]*)\(/\1\3(/g' "$TMP_FILE"
```

Comment:
We need to verify that nowhere have we put the `static` keyword on a separate line...

Reply:
Function name shouldn't contain spaces, but there could be whitespace after the function name.
```suggestion
sed -E -i 's/^([[:space:]]*)static([[:space:]]+)([a-zA-Z_][a-zA-Z0-9_:<>~\*&]*)([[:space:]]*)\(/\1\3(/g' "$TMP_FILE"
```

I can imagine a lot of ways this pattern could fail, it might be fragile. Perhaps consider if there's another way.

Reply:
There are functions where other keywords appear ahead of `static`, which this would not capture.

Reply:
I have modified the regex to handle almost all of the cases for function capture.

Here is the output file for `src/` directory where the regex did not replaced the static keyword: 
[remaining_static_src.txt](https://github.com/user-attachments/files/20637150/remaining_static_src.txt)

Here is the output file for `src/` directory where it applied all the regexes to replace static keyword:
[remove_static_src.diff.txt](https://github.com/user-attachments/files/20637155/remove_static_src.diff.txt)